### PR TITLE
Fix: Update publish pypi workflow to use hatch

### LINKED ISSUE #100: Set up a github action using hatch to release to PyPI
Automate the release to PyPI using hatch.

To approach this issue, consult the hatch documentation and recommended GitHub action.

This task will finish and close #82.

### PR DESCRIPTION
**Pull request by lwasser**
this closes #100 #82

NOTE: i haven't set this up before with hatch but because hatch is using setuptools_scm i suspect it's the same workflow. let's see how this runs.

### CODE REVIEW

File: .github/workflows/publish-pypi.yml [hunk 1]
```diff
@@ -45,6 +44,7 @@ jobs:
         if: github.event_name == 'push'
         uses: pypa/gh-action-pypi-publish@release/v1
         with:
+          # Haven't setup trusted workflow on test pypi yet
```
Comment (lwasser) [inline .github/workflows/publish-pypi.yml:47]: The weird part here is i'm not sure whether we want to use hatch publish vs just the pypi publish workflow which is easy and i've used before.

Comment (willingc) [reply .github/workflows/publish-pypi.yml:47]: I think it is fine to keep pypa/gh-action-pypi-publish

Comment (willingc) [reply .github/workflows/publish-pypi.yml:47]: Using hatch to build the wheel and sdist

**Comment by lwasser** [comment]
merging. we can see if it goes to testpypi

**Pull request merged**